File: solutions/python/general-algos/trees/invert_binary_tree.py

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None
        
        inverted_right_subtree = self.invertTree(root.right)
        inverted_left_subtree = self.invertTree(root.left)
        
        root.left = inverted_right_subtree
        root.right = inverted_left_subtree
        
        return root

File: solutions/javascript/general-algos/trees/invertBinaryTree.js
const invertTree = (root) => {
    if (!root) return null;
    
    const invertedRightSubtree = invertTree(root.right);
    const invertedLeftSubtree = invertTree(root.left);
    
    root.left = invertedRightSubtree;
    root.right = invertedLeftSubtree;
    
    return root;
};